rspace/src/main/scala/coop/rchain/rspace/RSpaceOps.scala

```diff
@@ -347,8 +347,8 @@ abstract class RSpaceOps[F[_]: Concurrent: ContextShift: Log: Metrics: Span, C,
     /*spanF.trace(createSoftCheckpointSpanLabel) */
     for {
       cache    <- storeAtom.get().snapshot
-      log      = eventLog.take()
-      _        = eventLog.put(Seq.empty)
+      log      <- eventLog.get
+      _        <- eventLog.update(_ => Seq.empty)
```

Comment:
Here this is not really critical because `createSoftCheckpoint` will always be called on single thread, but we want to ensure that read/write to Ref happens atomically.
`Ref` have different methods to update the state,  `set` ignores the existing state, `update` to change existing state and `modify` to change but also return some result. _Modify_ can replace all others, as _update_ can also cover _set_.

There are also combinations with `get` so in this case `getAndSet` fits perfectly, it will return existing value and set the new one.

Reply:
I fixed this